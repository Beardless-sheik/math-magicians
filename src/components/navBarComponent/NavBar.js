import cx from 'classnames';
import { Link } from 'react-router-dom';
import styles from './NavBar.module.css';

const NavBar = () => (
  <>
    <div className={styles.navContainer}>
      <h1> Math Magicians </h1>
      <div className={styles.navLinkCenter}>
        <ul className={styles.linkContainer}>
          <li className={cx(styles.linkItem)}>
            <Link to="/" className={styles.borderRightList}> Home </Link>
          </li>
          <li className={cx(styles.linkItem)}>
            <Link to="/calculator" className={styles.borderRightList}> Calculator </Link>
          </li>
          <li className={styles.linkItem}>
            <Link to="/quote"> Quote </Link>
          </li>
        </ul>
      </div>
    </div>
  </>
);

export default NavBar;
