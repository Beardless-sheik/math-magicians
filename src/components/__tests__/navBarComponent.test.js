import React from 'react';
import renderer from 'react-test-renderer';
import { HashRouter } from 'react-router-dom';
import NavBar from '../navBarComponent/NavBar';

describe('NavBar component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<HashRouter><NavBar /></HashRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
