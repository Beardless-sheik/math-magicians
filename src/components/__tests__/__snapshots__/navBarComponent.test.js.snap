// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator page component renders correctly 1`] = `
<div
  className="navContainer"
>
  <h1>
     Math Magicians 
  </h1>
  <div
    className="navLinkCenter"
  >
    <ul
      className="linkContainer"
    >
      <li
        className="linkItem"
      >
        <a
          className="borderRightList"
          href="#/"
          onClick={[Function]}
        >
           Home 
        </a>
      </li>
      <li
        className="linkItem"
      >
        <a
          className="borderRightList"
          href="#/calculator"
          onClick={[Function]}
        >
           Calculator 
        </a>
      </li>
      <li
        className="linkItem"
      >
        <a
          href="#/quote"
          onClick={[Function]}
        >
           Quote 
        </a>
      </li>
    </ul>
  </div>
</div>
`;

exports[`NavBar component renders correctly 1`] = `
<div
  className="navContainer"
>
  <h1>
     Math Magicians 
  </h1>
  <div
    className="navLinkCenter"
  >
    <ul
      className="linkContainer"
    >
      <li
        className="linkItem"
      >
        <a
          className="borderRightList"
          href="#/"
          onClick={[Function]}
        >
           Home 
        </a>
      </li>
      <li
        className="linkItem"
      >
        <a
          className="borderRightList"
          href="#/calculator"
          onClick={[Function]}
        >
           Calculator 
        </a>
      </li>
      <li
        className="linkItem"
      >
        <a
          href="#/quote"
          onClick={[Function]}
        >
           Quote 
        </a>
      </li>
    </ul>
  </div>
</div>
`;

exports[`NavBar page component renders correctly 1`] = `
<div
  className="navContainer"
>
  <h1>
     Math Magicians 
  </h1>
  <div
    className="navLinkCenter"
  >
    <ul
      className="linkContainer"
    >
      <li
        className="linkItem"
      >
        <a
          className="borderRightList"
          href="#/"
          onClick={[Function]}
        >
           Home 
        </a>
      </li>
      <li
        className="linkItem"
      >
        <a
          className="borderRightList"
          href="#/calculator"
          onClick={[Function]}
        >
           Calculator 
        </a>
      </li>
      <li
        className="linkItem"
      >
        <a
          href="#/quote"
          onClick={[Function]}
        >
           Quote 
        </a>
      </li>
    </ul>
  </div>
</div>
`;
